src/Containers.jsx

```diff
@@ -314,45 +316,58 @@ class Containers extends React.Component {
+        } else if (this.props.textFilter.length > 0) {
+            emptyCaption = emptyCaptionPod = _("No containers that match the current filter");
+        } else if (this.props.onlyShowRunning)
             emptyCaption = _("No running containers");
-
-        const containersStats = this.props.containersStats;
-        let filtered = [];
-        if (this.props.containers !== null)
-            filtered = Object.keys(this.props.containers).filter(id => !this.props.onlyShowRunning || this.props.containers[id].State == "running");
-
-        if (this.props.ownerFilter !== "all") {
-            filtered = filtered.filter(id => {
-                if (this.props.ownerFilter === "system" && !this.props.containers[id].isSystem)
-                    return false;
-                if (this.props.ownerFilter !== "system" && this.props.containers[id].isSystem)
-                    return false;
-                return true;
-            });
+            emptyCaptionPod = _("No running containers in this pod");
         }
 
-        if (this.props.textFilter.length > 0) {
-            const lcf = this.props.textFilter.toLowerCase();
-            filtered = filtered.filter(id => this.props.containers[id].Names[0].toLowerCase().indexOf(lcf) >= 0 ||
-                    this.props.containers[id].Image.toLowerCase().indexOf(lcf) >= 0
-            );
-        }
+        if (this.props.containers !== null && this.props.pods !== null) {
+            if (this.props.containers !== null && this.props.pods !== null)
+                filtered = Object.keys(this.props.containers).filter(id => !this.props.onlyShowRunning || this.props.containers[id].State == "running");
+
+            if (this.props.ownerFilter !== "all") {
+                filtered = filtered.filter(id => {
+                    if (this.props.ownerFilter === "system" && !this.props.containers[id].isSystem)
+                        return false;
+                    if (this.props.ownerFilter !== "system" && this.props.containers[id].isSystem)
+                        return false;
+                    return true;
+                });
+            }
 
-        filtered.sort((a, b) => {
-            // User containers are in front of system ones
-            if (this.props.containers[a].isSystem !== this.props.containers[b].isSystem)
-                return this.props.containers[a].isSystem ? 1 : -1;
-            return this.props.containers[a].Names > this.props.containers[b].Names ? 1 : -1;
-        });
+            if (this.props.textFilter.length > 0) {
+                const lcf = this.props.textFilter.toLowerCase();
+                filtered = filtered.filter(id => this.props.containers[id].Names[0].toLowerCase().indexOf(lcf) >= 0 ||
+                        this.props.containers[id].Image.toLowerCase().indexOf(lcf) >= 0
+                );
+            }
+
+            // Remove infra containers
+            filtered = filtered.filter(id => !this.props.containers[id].IsInfra);
```

Comment:
This seems like a weird place to do this. I think it should not be in the list of containers. Or would that overly-complicate things?

Reply:
This was delibatate. Some pods details like port mappings are present only in the infra Container details, and not in the Pod object itself.
Thus it makes sense to keep the infra container until this component, where we can get the details we are interested in.